### PR TITLE
[AF-913]: Added Categories to resource types

### PR DESCRIPTION
**Pull request by adrielparedes**
Cateogory added to ResourceTypeDefinitions. This way they can be filtered by category in assets view.

Issues related:
https://github.com/kiegroup/appformer/pull/116
https://github.com/kiegroup/kie-wb-common/pull/1317
https://github.com/kiegroup/drools-wb/pull/727
(new pr on designer)
https://github.com/kiegroup/jbpm-designer/pull/732

### CODE REVIEW

**Comment by adrielparedes** [comment]
jenkins retest this

**Comment by adrielparedes** [comment]
jenkins retest this

**Comment by adrielparedes** [comment]
jenkins retest this

**Comment by adrielparedes** [comment]
jenkins retest this

File: drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-client/src/main/java/org/drools/workbench/screens/dtablexls/client/type/DecisionTableXLSResourceType.java [hunk 1]
```diff
@@ -17,28 +17,38 @@
...
 @ApplicationScoped
 public class DecisionTableXLSResourceType
         extends DecisionTableXLSResourceTypeDefinition
         implements ClientResourceType {
 
+    public DecisionTableXLSResourceType() {
+    }
+
+    @Inject
+    public DecisionTableXLSResourceType(Decision category) {
```
Comment (manstis) [inline drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-client/src/main/java/org/drools/workbench/screens/dtablexls/client/type/DecisionTableXLSResourceType.java]: It'd be nice to be consistent with your use of ```final``` on method parameters (most occurrences in this PR are marked as ```final```)

File: drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-client/src/main/java/org/drools/workbench/screens/dtablexls/client/type/DecisionTableXLSXResourceType.java [hunk 2]
```diff
@@ -17,28 +17,38 @@
...
 @ApplicationScoped
 public class DecisionTableXLSXResourceType
         extends DecisionTableXLSXResourceTypeDefinition
         implements ClientResourceType {
 
+    public DecisionTableXLSXResourceType() {
+    }
+
+    @Inject
+    public DecisionTableXLSXResourceType(Decision category) {
```
Comment (manstis) [inline drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-client/src/main/java/org/drools/workbench/screens/dtablexls/client/type/DecisionTableXLSXResourceType.java]: ```final```

File: drools-wb-screens/drools-wb-globals-editor/drools-wb-globals-editor-client/src/main/java/org/drools/workbench/screens/globals/client/type/GlobalResourceType.java [hunk 3]
```diff
@@ -32,20 +33,30 @@
         extends GlobalResourceTypeDefinition
         implements ClientResourceType {
 
-    private static final Image IMAGE = new Image( GlobalsEditorResources.INSTANCE.images().typeGlobalVariable() );
+    private static final Image IMAGE = new Image(GlobalsEditorResources.INSTANCE.images().typeGlobalVariable());
 
-    @Inject
     private TranslationService translationService;
 
+    public GlobalResourceType(Decision category) {
```
Comment (manstis) [inline drools-wb-screens/drools-wb-globals-editor/drools-wb-globals-editor-client/src/main/java/org/drools/workbench/screens/globals/client/type/GlobalResourceType.java]: ```final```

File: drools-wb-screens/drools-wb-globals-editor/drools-wb-globals-editor-client/src/main/java/org/drools/workbench/screens/globals/client/type/GlobalResourceType.java [hunk 4]
```diff
@@ -32,20 +33,30 @@
...
+        super(category);
+    }
+
+    @Inject
+    public GlobalResourceType(TranslationService translationService,
```
Comment (manstis) [inline drools-wb-screens/drools-wb-globals-editor/drools-wb-globals-editor-client/src/main/java/org/drools/workbench/screens/globals/client/type/GlobalResourceType.java]: ```final```

File: drools-wb-screens/drools-wb-globals-editor/drools-wb-globals-editor-client/src/main/java/org/drools/workbench/screens/globals/client/type/GlobalResourceType.java [hunk 5]
```diff
@@ -32,20 +33,30 @@
...
+                              Decision category) {
```
Comment (manstis) [inline drools-wb-screens/drools-wb-globals-editor/drools-wb-globals-editor-client/src/main/java/org/drools/workbench/screens/globals/client/type/GlobalResourceType.java]: ```final```

Comment (adrielparedes) [reply drools-wb-screens/drools-wb-globals-editor/drools-wb-globals-editor-client/src/main/java/org/drools/workbench/screens/globals/client/type/GlobalResourceType.java]: @manstis done all changes, thanks for the review!

**Pull request merged**